src/GameLoop.cpp

```diff
@@ -40,6 +40,10 @@ void GameLoop::SetUpdateRate( float fUpdateRate )
 	g_fUpdateRate = fUpdateRate;
 }
 
+float& GameLoop::GetUpdateRate() {
```

Comment:
If you want to return the actual instance of the rate, make sure to mark the return value as as `const float&` so that callers can't inadvertently set the rate from outside the class.

As it is something like this would update `g_fUpdateRate`

```cpp
int& foo = GameLoop::GetUpdateRate();
foo = 10000;
```

`g_fUpdateRate` would also get set to 10000 then

I would actually add a comment why we're returning the actual instance of the float instead of a copy (which would be more normal). (Something akin to "actors are very common so incurring a copy cost, albeit minor, doesn't seem necessary. We return a const reference to ensure callers can't modify the underlying variable.)

Reply:
Good catch, fixed, and I added a comment.